### PR TITLE
Long term dashboard

### PR DESCRIPTION
**Pull request by olliejth**
Added functionality to add visualisations to the streamlit dashboard that are created from the long-term s3 info.

### CODE REVIEW

File: dashboard/load_dashboard.py [hunk 1]
```diff
@@ -6,9 +6,14 @@
 from extract_dashboard import extract_readings
 from transform_dashboard import create_botanist_pie, create_temperature_bar
 from transform_dashboard import create_moisture_bar, create_temperature_line
+from longterm_extract import extract_s3_files
+from longterm_transform import concat_csvs, get_times_watered, get_max_temp_per_plant, get_min_moisture_per_plant
+
 
 if __name__ == "__main__":
 
+    # SHORT TERM VISUALISATIONS
```
Comment (mehrdadhalali) [inline dashboard/load_dashboard.py:15]: Good use of comments, Mehrdad approves.

File: dashboard/load_dashboard.py [hunk 2]
```diff
@@ -27,3 +32,15 @@
     st.altair_chart(temp_bar_chart, use_container_width=True)
     st.altair_chart(temp_line_chart, use_container_width=True)
     st.altair_chart(moist_bar_chart, use_container_width=True)
+
+    # LONG TERM VISUALISATIONS:
+
+    longterm_df = concat_csvs(extract_s3_files())
+
+    water_count_bar = get_times_watered(longterm_df)
+    max_temp_bar = get_max_temp_per_plant(longterm_df)
```
Comment (Leon-JS02) [inline dashboard/load_dashboard.py:41]: Very neat dashboarding code. For a future improvement we could consider using Altair's columns to divide it.

File: dashboard/longterm_transform.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    title = alt.TitleParams(
+        'Min M experienced by each plant', anchor='middle')
+    min_moist_bar = alt.Chart(min_moist_df, title=title).mark_bar().encode(
+        x="plant_id:N",
+        y="min_M:Q").configure_bar(
+        color='red'
+    )
+
+    return min_moist_bar
```
Comment (Leon-JS02) [inline dashboard/longterm_transform.py:66]: Nice.

**Review by Leon-JS02 — APPROVED** [review]
Neat dashboarding code. Very concise.

File: dashboard/longterm_transform.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    return bar_chart
+
+
+def get_max_temp_per_plant(df: pd.DataFrame) -> alt.Chart:
+
+    max_temp_df = df.groupby("plant_id").max().reset_index()
+
+    title = alt.TitleParams(
+        'Max T experienced by each plant', anchor='middle')
```
Comment (mehrdadhalali) [inline dashboard/longterm_transform.py:44]: "Max plant temperature"?

**Review by mehrdadhalali — APPROVED** [review]
Mehrproved

**Pull request merged**